aten/src/ATen/native/SharedReduceOps.h

```diff
@@ -21,9 +21,26 @@
 #define device_sqrt std::sqrt
 #endif
 #if defined(__CUDACC__) || defined(__HIPCC__)
-#define MAX(X, Y) ::max(X,Y)
-#define MIN(X, Y) ::min(X,Y)
+template <typename scalar_t>
+inline C10_DEVICE scalar_t max_propagate_nan(scalar_t a, scalar_t b) {
+  if (at::_isnan(a) || at::_isnan(b)) {
+    return std::numeric_limits<scalar_t>::quiet_NaN();
+  } else {
+    return ::max(a, b);
+  }
+}
+template <typename scalar_t>
+inline C10_DEVICE scalar_t min_propagate_nan(scalar_t a, scalar_t b) {
+  if (at::_isnan(a) || at::_isnan(b)) {
```

Comment:
slightly simpler way is
```
scalar_t min = std::isnan(b) ? b : std::min(a,b); //std::min returns first arg if one of the args is nan
```

Reply:
I'm getting some ROCm failures and I think (not 100% sure) that this is the culprit: https://github.com/ROCm-Developer-Tools/HIP/issues/2209

I'll add the isnan check of `a` for back in for HIP, hopefully that fixes it.

Reply:
Confirmed, adding an `at::_isnan(a)` check for HIP fixed the failures. I added a comment with a link to the above HIP issue